util/exec/exec.go

```diff
@@ -41,5 +42,14 @@ func RunWithRedactor(cmd *exec.Cmd, redactor func(text string) string) (string,
 		span.SetBaggageItem("args", fmt.Sprintf("%v", cmd.Args))
 	}
 	defer span.Finish()
-	return argoexec.RunCommandExt(cmd, opts)
+	output, err := argoexec.RunCommandExt(cmd, opts)
+	if err != nil {
+		re, ok := err.(*argoexec.CmdError)
```

Comment:
I'm torn on this request but... do you think it'd be helpful to have a unit test to demo the behavior?

Reply:
I can add one to prevent regressions, yes

Reply:
@crenshaw-dev updated with a unittest

Reply:
I think the red error dialog can definitely be improved. Is there a guarantee that only the first 80 characters in the message are relevant, and nothing beyond that isn't?  Perhaps from other commands other than helm? One possible suggestion is leave this code as is, and  just enhance that error dialog to perhaps show the first 80 characters, and then provide a 'More' button to show the rest of the text?  What do you think?

Reply:
@keithchong this change is not triming the first 80 characters _of the error message_. Instead, it is aware when the error was produced by a CLI call in the backend, and trims the first 80 characters of the executed CLI (could be helm, kustomize, plugin etc). The rest of the error message is shown intact - as raised in the code.

The full CLI call are printed in the logs, just like before.


This is particularly useful for helm (which uses more than 500 chars in CLI to specify various apiGroups), but I would deem it is harmless for other usages, as it prints the error in full (trimming only CLI arguments).

I agree we can also do more in the frontend, however in the frontend we have no idea which part of the error is the CLI call and which part is the actual error being raised - unless we do some string magic which might get ugly.

Reply:
Hi @alexef , thanks for the clarification. I looked at 'truncate error' part of this PR abstract and wondered if that was the intended change. And this PR addresses Proposal 2 in the issue. 

The issue I was thinking about was https://github.com/argoproj/argo-cd/issues/4901.  (That popup definitely can be improved.)

However, IMHO, this seems like a band-aid fix in place of the overall needed usability fix in the UI that will be addressed by proposal 1.  i agree with your last point, but maybe that could be part of the proposal 1 solution. Just eyeballing the argoproj's exec package...maybe something can be done there to separate out the command and args from the error message. I could be wrong but it might be the culprit that combines them.

@crenshaw-dev , what do you think?

Reply:
@alexmt  WDYT?

Reply:
@keithchong looking at #4901, I'm gonna open a separate PR for it, frontend only.